src/main/java/ch/njol/skript/registrations/EventValues.java

```diff
@@ -19,224 +19,351 @@
 package ch.njol.skript.registrations;
 
 import java.util.ArrayList;
-import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 import org.bukkit.event.Event;
 import org.eclipse.jdt.annotation.Nullable;
 
 import com.google.common.collect.ImmutableList;
+
 import ch.njol.skript.Skript;
 import ch.njol.skript.classes.Converter;
 import ch.njol.skript.expressions.base.EventValueExpression;
 import ch.njol.skript.util.Getter;
 
 /**
- * @author Peter Güttinger
+ * The main class for registering and getting anything relating to event values.
  */
 public class EventValues {
-	
-	private EventValues() {}
-	
-	private final static class EventValueInfo<E extends Event, T> {
-		
-		public final Class<E> event;
-		public final Class<T> c;
-		public final Getter<T, E> getter;
-		@Nullable
-		public final Class<? extends E>[] excludes;
+
+	private final static class EventValueInfo<E extends Event, T> implements Comparable<EventValueInfo<?, ?>> {
+
+		private final Class<? extends E>[] excludes;
+
 		@Nullable
-		public final String excludeErrorMessage;
-		
-		public EventValueInfo(Class<E> event, Class<T> c, Getter<T, E> getter, @Nullable String excludeErrorMessage, @Nullable Class<? extends E>[] excludes) {
-			assert event != null;
-			assert c != null;
+		private final String excludeErrorMessage;
+
+		private final Getter<T, E> getter;
+		private final Class<E> event;
+		private final Class<T> type;
+
+		private int priority;
+
+		@SuppressWarnings("unchecked")
+		public EventValueInfo(Class<E> event, Class<T> type, Getter<T, E> getter, int priority, @Nullable String excludeErrorMessage, @Nullable Class<? extends E>... excludes) {
 			assert getter != null;
-			this.event = event;
-			this.c = c;
-			this.getter = getter;
-			this.excludes = excludes;
+			assert event != null;
+			assert type != null;
+
+			this.excludes = excludes == null ? new Class[0] : excludes;
+			if (excludes.length > 0 && excludeErrorMessage == null)
+				Skript.error("The exclude error message cannot be null when there are excluded events.");
+
 			this.excludeErrorMessage = excludeErrorMessage;
+			this.priority = priority;
+			this.getter = getter;
+			this.event = event;
+			this.type = type;
 		}
-		
-		/**
-		 * Get the class that represents the Event.
-		 * @return The class of the Event associated with this event value
-		 */
-		public Class<E> getEventClass() {
-			return event;
-		}
-		
-		/**
-		 * Get the class that represents Value.
-		 * @return The class of the Value associated with this event value
-		 */
-		public Class<T> getValueClass() {
-			return c;
+
+		@Override
+		public int compareTo(EventValueInfo<?, ?> other) {
+			return Integer.compare(Math.abs(priority), Math.abs(other.priority));
```

Comment:
Instead of worrying about cleaning numbers here, maybe an IAE if the priority is below 0 during construction

Reply:
What's an IAE?

Reply:
Priority in the negatives can be used by addons to override Skript's priority as Skript never uses negative numbers,

Reply:
If you take the absolute value though, then an addon can't really override Skript can it?

If Skript has a priority of 10 and my addon's is -20, Skript wins after absolute value calculations